src/ACEtk/block/details/BaseDistributionData/src/generateXSS.hpp

```diff
@@ -6,19 +6,19 @@ generateXSS( const std::string& name,
              std::size_t locb ) {
 
   // add the distribution data to the xss array
-  std::size_t nr = boundaries.size();
-  std::size_t ne = distributions.size();
-  std::vector< double > xss( 1 + 2 * ne );
+  const auto nr = boundaries.size();
+  const auto ne = distributions.size();
+  std::vector< double > xss( static_cast< double >( 1 + 2 * ne ) );
```

Comment:
This appears in a few places, `xss` is constructed using a double instead of what previously was an int/size_t, which is passed to the size. I think that's the only thing I've seen in this PR that is an issue.

Reply:
Approved once these are all fixed.

Reply:
I did not pay attention here it would appear. This should only be done in the initialiser list, e.g. 
```
xss = { static_cast<double>(...) }
```
I can fix this.

Reply:
I found 7 places where this had happened. These are fixed now.